Sources/Optimizely+Decide/OptimizelyUserContext.swift

```diff
@@ -0,0 +1,71 @@
+/****************************************************************************
+* Copyright 2020, Optimizely, Inc. and contributors                        *
+*                                                                          *
+* Licensed under the Apache License, Version 2.0 (the "License");          *
+* you may not use this file except in compliance with the License.         *
+* You may obtain a copy of the License at                                  *
+*                                                                          *
+*    http://www.apache.org/licenses/LICENSE-2.0                            *
+*                                                                          *
+* Unless required by applicable law or agreed to in writing, software      *
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+    
+
+import Foundation
+
+/// A struct for user contexts that the SDK will use to make decisions for.
+public struct OptimizelyUserContext {
+    weak var optimizely: OptimizelyClient?
```

Comment:
since this is weak and could go out of scope, we can add a guard error now for optimizely is nil

Reply:
A good point. Will add the guard and tests in other PRs (#347, #348).